app/models/cafe_work.rb

```diff
@@ -0,0 +1,252 @@
+      return false
+    end
+
+    # Should be done with a bang when the error handling works
+    # Ref: https://github.com/fsek/web/issues/93
+    # /d.wessman
+    self.attributes = worker_params
+    self.attributes = Assignee.setup(worker_attributes,user).attributes
+    save
+  end
+
+  # Remove-function used by the worker
+  # /d.wessman
+  def remove_worker(user, access)
+    if (!owner?(user) && !authorize(access))
+      errors.add('Auktorisering', 'misslyckades, du har inte rättighet att ta bort eller skrev fel kod.')
+      return false
+    end
+
+    clear_worker
+  end
+
+  # Method to remove the worker from current work.
+  # /d.wessman
+  def clear_worker
+    self.attributes = worker.clear_attributes
+    self.utskottskamp = false
+    self.councils.clear
+    self.save!(validate: false)
+  end
+
+  # Returns true if the profiles are similar and not nil
+  # /d.wessman
+  def owner?(user)
+    user.present? && user.profile.present? && profile.present? && user.profile == profile
+  end
+
+  # Returns true if the user can edit the object
+  # /d.wessman
+  def edit?(user)
+    editable? && (!has_worker? || owner?(user))
+  end
+
+  def editable?
+    work_day > Time.zone.now
+  end
+
+  # Returns true only if hte access_code is correct
+  # /d.wessman
+  def authorize(access)
+    access.present? && access_code.present? && access_code == access
+  end
+
+  # Returns true if there is a worker
+  # /d.wessman
+  def has_worker?
+    worker.present?
+  end
+
+  # Used to print date in a usable format.
+  # /d.wessman
+  def print_time
+    %(#{start.strftime("%H:%M")}-#{stop.strftime("%H:%M")})
+  end
+
+  # Used to print out date, reading week and work number
+  # /d.wessman
+  def print
+    %(#{print_date} LV: #{lv} Pass: #{pass})
+  end
+
+  def print_date
+    %(#{print_time}, #{start.strftime("%A %d/%m")})
+  end
+
+  # Prints the url or path for the current object
+  def p_url
+    Rails.application.routes.url_helpers.cafe_work_url(id, host: PUBLIC_URL)
+  end
+
+  def p_path
+    Rails.application.routes.url_helpers.cafe_work_path(id)
+  end
+
+  def as_json(*)
+    {
+        id: id,
+        title: %(Cafepass #{pass}),
+        start: start.iso8601,
+        end: stop.iso8601,
+        status: print,
+        url: p_path,
+        color: 'black',
+        backgroundColor: b_color,
+        textColor: 'black'
+    }
+  end
+
+  # To print start time
+  def t_start
+    start.strftime('%H:%M')
+  end
+
+  # To print end time
+  def t_end
+    stop.strftime('%H:%M')
+  end
+
+  def start
+    work_day
+  end
+
+  # End would be a better name, doesn't fit into code.
+  def stop
+    work_day + duration.hours
+  end
+
+
+  def self.get_lv
+    check = CafeWork.between(Time.zone.now.beginning_of_day-2.days, Time.zone.now.end_of_day).last
+    (check.present?) ? check.lv.to_s : '?'
+  end
+
+  protected
+
+  def u_attributes(params)
```

Comment:
Unused method argument - `params`. If it's necessary, use `_` or `_params` as an argument name to indicate that it won't be used. You can also write as `u_attributes(*)` if you want the method to accept any arguments but don't care about them.